src/modules/site-v2/base/views/tools/phenotype_database.py

```diff
@@ -59,12 +63,89 @@ def results_columns():
 @phenotype_database_bp.route('')
 @cache.memoize(60*60)
 def phenotype_database():
+  """
+    Phenotype Database table (non-bulk)
+  """
+  # Get the list of traits for non-bulk files
+  try:
+    traits_non_bulk = query_phenotype_metadata()
+  except Exception as ex:
+    logger.error(f'Failed to retrieve the list of traits: {ex}')
+    abort(500)
+
   return render_template('tools/phenotype_database/phenotypedb.html', **{
     # Page info
     "title": 'Phenotype Database and Analysis',
     "tool_alt_parent_breadcrumb": { "title": "Tools", "url": url_for('tools.tools') },
+
+    # Data
+    'traits': traits_non_bulk,
   })
 
+@phenotype_database_bp.route('/traits-zhang')
+@cache.memoize(60*60)
+@compress.compressed()
+def get_zhang_traits_json():
+  """
+    Phenotype Database table (bulk)
+    Fetch table content by request for each page and render the data for Datatables()
+  """
+  try:
+    # get parameters for query
+    draw = request.args.get('draw', type=int)
+    start = request.args.get('start', type=int)
+    length = request.args.get('length', type=int)
+    search_value = request.args.get('search[value]', '').lower()
```

Comment:
we should add a guard here to validate *all* the user input and assert that it is in valid ranges/values/lists/etc. If the request can't be processed, we should catch it before tying to process it and notify the user with a proper error (e.g.: 400 bad request)>